tests/main/snap-advise-command/task.yaml

```diff
@@ -0,0 +1,27 @@
+summary: Ensure that `snap advise-command` works
+
+execute: |
+    echo "wait for snapd to pull in the commands data"
+    echo "(it will do that on startup)"
+    for i in $(seq 120); do
+       if stat /var/cache/snapd/commands.db; then
```

Comment:
Is it possible that we'll continue even though the bucket is still unpopulated? IIRC, the setup of boltFinder will first create an empty bucket and commit, probably thus creating a file, however that does not mean that the command-not-found data has been populated yet.

An alternative would be to dump the contents of db to make sure that there's something inside. Given that boltdb does not provide a cli anymore, we'd need to use this small piece of Go code:
```
package main

import (
	"errors"
	"fmt"
	"log"
	"os"

	"github.com/boltdb/bolt"
)

func main() {
	if len(os.Args) != 2 {
		log.Fatalf("usage: get <db-path>")
	}
	path := os.Args[1]
	db, err := bolt.Open(path, 0600, &bolt.Options{ReadOnly: true})
	if err != nil {
		log.Fatalf("failed to open DB: %v", err)
	}
	err = db.View(func(tx *bolt.Tx) error {
		b := tx.Bucket([]byte("Commands"))
		if b == nil {
			return errors.New("no bucket")
		}
		c := b.Cursor()
		for k, v := c.First(); k != nil; k, v = c.Next() {
			fmt.Printf("%v: %v\n", string(k), string(v))
		}
		return nil
	})
	if err != nil {
		log.Fatalf("dump failed: %v", err)
	}
}
```
and `test "$(get /var/cache/snapd/commands.db | wc -l)" -gt 10`. Maybe even keep it in `tools/cmd/boltdump` in case it proves useful in the future.

Reply:
Hm, interessting idea - I think we should probably go back the original code that populates the bucket and make sure we only create it when we have data in it.